pkg/providers/amifamily/ubuntu.go

```diff
@@ -52,7 +54,7 @@ func (u Ubuntu) DefaultAMIs(version string) []DefaultAMIOutput {
 }
 
 // UserData returns the default userdata script for the AMI Family
-func (u Ubuntu) UserData(kubeletConfig *v1alpha5.KubeletConfiguration, taints []v1.Taint, labels map[string]string, caBundle *string, _ []*cloudprovider.InstanceType, customUserData *string) bootstrap.Bootstrapper {
+func (u Ubuntu) UserData(kubeletConfig *v1alpha5.KubeletConfiguration, _ *version.Version, taints []v1.Taint, labels map[string]string, caBundle *string, _ []*cloudprovider.InstanceType, customUserData *string) bootstrap.Bootstrapper {
```

Comment:
AFAIK Ubuntu uses the same eks-bootstrap script that contains the removal of the `--container-runtime` parameter so we should check if this is also something that we should remove from this AMIFamily as well.

Reply:
It didn't look like any AMI types other than AL2 did, I can take a look and make sure though. The reason I didn't remove it for AMIFamily as a whole is `dockerd` is still supported by K8s 1.23.